pkg/cloud/azure/azure_storage.go

```diff
@@ -157,6 +171,36 @@ func (s *azureStorage) Writer(ctx context.Context, basename string) (io.WriteClo
 	}), nil
 }
 
+type putBlobUploader struct {
+	ctx  context.Context
+	blob azblob.BlockBlobURL
+	b    *bytes.Buffer
+	sp   *tracing.Span
+}
+
+func (u *putBlobUploader) Write(p []byte) (int, error) {
+	return u.b.Write(p)
+}
+
+func (u *putBlobUploader) Close() error {
+	defer u.sp.Finish()
```

Comment:
^somebody needs to call this, no? Maybe in `(s *azureStorage) Close()`

Reply:
I believe the caller of Writer is responsible for closing the returned writer.